### ISSUE TITLE
Usage document needed.

### ISSUE THREAD

**Issue by johnking** [jdalrymple/node-gitlab#172]
Hi @jdalrymple ,

first of all, thanks a lot for the package. 

As a new user, I found it bit of challenge to use the library, for example, 

1. Based on the example code in README.md, by calling `api.projects.all()` always get a `{"statusCode":500,"error":"Internal Server Error","message":"An internal server error occurred"}`, until I added some query parameters like  `api.projects.all({membership: true})` to make it working, it took me quite some time to figure it out.

2. Tried to use `api.ProjectMembers.all()`, my first try is `api.ProjectMembers.all({id: xxxxx})` and again got the same `500 error`. Until later I found I should use api.ProjectMembers.all(xxxxxx).

May you please provide some guidances on the API calls so people can understand the usage easily? 

thanks again.

**Comment by johnking** [comment]
By referring to the source code, I can understand the usages now. 

thanks anyway.

**Comment by jdalrymple** [comment]
Hey @johnking ! We definitely need some form of documentation, its something weve been thinking about for a while and is being discussed in #39. As for the troubles you where having, the all function should work without parameters so I'm unsure as to why you got an error 500 :s. The options however that you can pass into the all function are any options that gitlab supports for the projects/ endpoint.

**Comment by johnking** [comment]
Hi @jdalrymple , thanks for your reply. For the `500 errors`, I were also puzzled since the code were so simple, ( a few lines only), I tried both oauthToken and personal token, all gave me `500 error` unless I specify some queries. 

I think it might be due to the large amount of records returned or the timeout settings.

**Comment by mingyuan-xia** [comment]
Some very basic examples would get ppl a quick start.

**Comment by casz** [comment]
Think typescript types will help too which will be worked on once `next` branch is merged into master 👍

**Comment by johannessteu** [comment]
Just used this package for the first time. No 500er for me at all so far!

**Comment by jdalrymple** [comment]
Just going to reference this in the documentation issue and close this!
